File: seed/php-sdk/literal/src/Headers/Requests/SendLiteralsInHeadersRequest.php
<?php

namespace Seed\Headers\Requests;

use Seed\Core\Json\JsonSerializableType;
use Seed\Core\Json\JsonProperty;

class SendLiteralsInHeadersRequest extends JsonSerializableType
{
    /**
     * @var '02-12-2024' $endpointVersion
     */
    public string $endpointVersion;

    /**
     * @var true $async
     */
    public bool $async;

    /**
     * @var string $query
     */
    #[JsonProperty('query')]
    public string $query;

    /**
     * @param array{
     *   endpointVersion: '02-12-2024',
     *   async: true,
     *   query: string,
     * } $values
     */
    public function __construct(
        array $values,
    )
    {
        $this->endpointVersion = $values['endpointVersion'];$this->async = $values['async'];$this->query = $values['query'];
    }
}

File: seed/ruby-sdk-v2/literal/lib/seed/headers/types/send_literals_in_headers_request.rb
# frozen_string_literal: true

module Seed
  module Headers
    module Types
      class SendLiteralsInHeadersRequest < Internal::Types::Model
        field :endpoint_version, -> { String }, optional: false, nullable: false
        field :async, -> { Internal::Types::Boolean }, optional: false, nullable: false
        field :query, -> { String }, optional: false, nullable: false
      end
    end
  end
end
